lib/libutils/isoc/bget_malloc.c

```diff
@@ -609,7 +613,9 @@ void *mdbg_malloc(const char *fname, int lineno, size_t size)
 {
 	struct mdbg_hdr *hdr;
 
-	COMPILE_TIME_ASSERT(sizeof(struct mdbg_hdr) == sizeof(uint32_t) * 4);
+	/* check struct mdbg_hdr doesn't get bad alignment */
+	COMPILE_TIME_ASSERT(
+		(sizeof(struct mdbg_hdr) % __alignof(uintptr_t)) == 0);
```

Comment:
To get alignment 8 for ARM32 and 16 for ARM64 multiply `__alignof(uintptr_t)` with 2.